python/bind_data_access.h

```diff
@@ -192,22 +192,31 @@ template <class... Ts> class as_ElementArrayViewImpl {
                                                   scipp::core::time_point>(
           view);
     return std::visit(
-        [&view, &obj](const auto &data) {
+        [&view](const auto &data) {
+          const auto get_parent = [](auto &&view_) {
+            if constexpr (std::is_same_v<std::decay_t<decltype(view)>,
+                                         scipp::Variable>) {
+              return py::cast(view_.data_handle());
+            } else {
+              return py::cast(view_.data().data_handle());
+            }
+          };
```

Comment:
Can we reuse the helper we added in the other PR?

Reply:
Sure

Reply:
Like this?